moviepy/Clip.py

```diff
@@ -490,10 +497,103 @@ def close(self):
+
+    def __getitem__(self, key):
+        """
+        Support extended slice and index operations over 
+        a clip object. 
+        
+        Simple slicing is implemented via :meth:`subclip`. 
+        So, ``clip[t_start:t_end]`` is equivalent to  
+        ``clip.subclip(t_start, t_end)``. If ``t_start`` is not 
+        given, default to ``0``, if ``t_end`` is not given, 
+        default to ``self.duration``.   
+        
+        The slice object optionally support a third argument as 
+        a ``speed`` coefficient (that could be negative), 
+        ``clip[t_start:t_end:speed]``. 
+
+        For example ``clip[::-1]`` returns a reversed (a time_mirror fx)
+        the video and ``clip[:5:2]`` returns the segment from 0 to 5s
+        accelerated to 2x (ie. resulted duration would be 2.5s)  
+    
+        In addition, a tuple of slices is supported, resulting in the concatenation
+        of each segment. For example ``clip[(:1, 2:)]`` return a clip
+        with the segment from 1 to 2s removed.  
+
+        If ``key`` is not a slice or tuple, we assume it's a time 
+        value (expressed in any format supported by :func:`cvsec`)
+        and return the frame at that time, passing the key 
+        to :meth:`get_frame`. 
+        """
+        if isinstance(key, slice):
+            # support for [start:end:speed] slicing. If speed is negative
+            # a time mirror is applied.
+            clip = self.subclip(key.start or 0, key.stop or self.duration)
+
+            if key.step:
+                # change speed of the subclip
+                apply_to = self.__apply_to(clip)
+                factor = abs(key.step)
+                if factor != 1:
+                    # change speed
+                    clip = clip.fl_time(
+                        lambda t: factor * t, apply_to=apply_to, keep_duration=True
+                    )
+                    clip = clip.set_duration(1.0 * clip.duration / factor)
+                if key.step < 0:
+                    # time mirror
+                    clip = clip.fl_time(
+                        lambda t: clip.duration - t,
+                        keep_duration=True,
+                        apply_to=apply_to,
+                    )
+            return clip
+        elif isinstance(key, tuple):
+            # get a concatenation of subclips
+            return reduce(add, (self[k] for k in key))
+        else:
+            return self.get_frame(key)
+
+    def __del__(self):
+        self.close()
+
+    def __add__(self, other):
+        # concatenate. implemented in specialized classes
+        self.__unsupported(other, "+")
+
+    def __mul__(self, n):
+        # loop n times
+        if not isinstance(n, Real):
+            self.__unsupported(n, "*")
+
+        apply_to = self.__apply_to(self)
+        clip = self.fl_time(
+            lambda t: t % self.duration, apply_to=apply_to, keep_duration=True
+        )
+        return clip.set_duration(clip.duration * n)
```

Comment:
Any reason not to just do `return loop(clip, n=n)`? And if not, then you should use the `@requires_duration` decorator here as well I think.